tests/workspaces.rs

```diff
@@ -0,0 +1,684 @@
+extern crate cargotest;
+extern crate hamcrest;
+
+use cargotest::support::{project, execs};
+use cargotest::support::registry::Package;
+use hamcrest::{assert_that, existing_file, is_not};
+
+#[test]
+fn simple_explicit() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [project]
+            name = "foo"
+            version = "0.1.0"
+            authors = []
+
+            [workspace]
+            members = ["bar"]
+        "#)
+        .file("src/main.rs", "fn main() {}")
+        .file("bar/Cargo.toml", r#"
+            [project]
+            name = "bar"
+            version = "0.1.0"
+            authors = []
+            workspace = ".."
```

Comment:
Oh the `[package]` and `[project]` keys are actually synonymous right now. There were initial ideas of how to treat them differently but that never came to fruition, so it's it's just a semi-unfortunate legacy "feature"